verification/verifier_helpers.go

```diff
@@ -0,0 +1,53 @@
+package verification
+
+import (
+	"crypto/x509"
+	"fmt"
+	"strings"
+)
+
+// isFailureResult checks whether a VerificationResult fails the entire signature verification workflow.
+// signature verification workflow is considered failed if there is a VerificationResult with "Enforced" as the action but the result was inconclusive (failed to verify) or unsuccessful
+func isFailureResult(result VerificationResult) bool {
+	return result.Action == Enforced && (result.FailedToVerify || !result.Success)
+}
+
+func verifyX509TrustedIdentities(certs []*x509.Certificate, trustPolicy *TrustPolicy) error {
```

Comment:
The trusted identity (anchor) is checked against the signing certificate, can we pass only the signing cert instead of the chain.

Reply:
We are already using the leaf for this purpose - https://github.com/notaryproject/notation-go/blob/main/verification/verifier.go#L71

Since we are using pointers, there is no impact on performance either ways. I can still do it if required.

Reply:
Agreed, the suggestion is mainly to keep the method signature idiomatic, passing the whole chain allows a future change to incorrectly access other elements of the cert chain, whereas we want to restrict to only the leaf.